Framework/Core/src/runDataProcessing.cxx

```diff
@@ -874,9 +874,11 @@ bool processSigChild(DeviceInfos& infos, DeviceSpecs& specs)
         } else {
           if (WIFSIGNALED(status)) {
             int exitSignal = WTERMSIG(status);
-            LOGP(error, "Workflow crashed - pid {} ({}) was killed abnormally with {} and exited with {}", pid, id, strsignal(exitSignal), es);
+            es = exitSignal + 128;
+            LOGP(error, "Workflow crashed - PID {} ({}) was killed abnormally with {} and exited code was set to {}.", pid, id, strsignal(exitSignal), es);
           } else {
-            LOGP(error, "pid {} ({}) crashed with or was killed with exit code {}", pid, id, es);
+            es = 128;
+            LOGP(error, "PID {} ({}) did not exit correctly however it's not clear why. Exit code forced to {}.", pid, id, es);
```

Comment:
Shouldn't we have `Workflow crashed - ..." as well here, since it was not a clean shutdown, so that one can easily grep for it?

Reply:
I can change that. In principle it's not clear to me if WIFEXITED() == false and WIFSIGNALED() == false is actually possible at all.